code/controllers/ModelAsController.php

```diff
@@ -49,82 +49,79 @@ public function init() {
 	public function handleRequest(SS_HTTPRequest $request, DataModel $model) {
 		$this->setRequest($request);
 		$this->setDataModel($model);
-		
-		$this->pushCurrent();
 
-		// Create a response just in case init() decides to redirect
-		$this->response = new SS_HTTPResponse();
```

Comment:
If init is relying on `$this->response` then it would have to be calling `$this->getResponse()` which is an API change I didn't account for :\

Should I add in a local `$response` variable to this code or submit this further up the chain @kinglozzer ?